### PR TITLE
TEMPO/TEMPO2 compatibility

### PR DESCRIPTION
**Pull request by aarchiba**
The goal of this PR is to improve compatibility of written par/tim files with tempo/tempo2. In so far as possible normal PINT par/tim files should be intelligible to these programs, but it should be reasonably easy to ask PINT to produce more-compatible par/tim files and it should still be able to read them. This will be accomplished partly by tweaking various settings to PINT produces reasonably compatible output by default, but also by allowing translation of parameter names to tempo/tempo2-friendlier versions (that are, ideally, recognized as aliases upon reading).

This should produce TEMPO-friendly par and tim files, if the model is appropriate, with:
```python
to.alias_translation = pint.toa.tempo_aliases
mo.use_aliases(reset_to_default=True, alias_translation=pint.toa.tempo_aliases)
to.write_tim_file("file.tim")
with open('file.par', 'w') as f:
    f.write(mo.as_parfile())
```

Should help with #1008

### CODE REVIEW

File: src/pint/toa.py [hunk 1]
```diff
@@ -73,6 +73,16 @@
 
 all_planets = ("jupiter", "saturn", "venus", "uranus", "neptune", "earth")
 
+tempo_aliases = {
+    "arecibo": "AO",
+    "jodrell": "JB",
+    "A1DOT": "XDOT",
+    "STIGMA": "VARSIGMA",
+    "EFAC": "T2EFAC",
+    "EQUAD": "T2EQUAD",
```
Comment (scottransom) [inline src/pint/toa.py:82]: This is from memory, but I'm pretty sure this is correct, but T2EFAC and T2EQUAD aren't really aliases for EFAC and EQUAD, right?  Their order of ops is different.  I'll see if I can't find the documentation for that....

Comment (scottransom) [reply src/pint/toa.py:82]: Hmmm.  After looking at the TEMPO and T2 codes, maybe I'm wrong and they both are implemented the same.  I swear this was a problem sometime in the past, but maybe it was an isolated incident or something.

**Comment by swiggumj** [comment]
I was scrolling through various issues that came up and wondered if this PR would also handle the fact that TEMPO expects DMDATA and NHARM to be integers. Can we simply make that so in PINT? And from a 15yr (timing_analysis) standpoint, to facilitate PINT/Tempo2 comparisons, should we specify IERS2003 as an early check (I can open a quick MR if so) for timing models?

**Comment by aarchiba** [comment]
> I was scrolling through various issues that came up and wondered if this PR would also handle the fact that TEMPO expects DMDATA and NHARM to be integers. Can we simply make that so in PINT? And from a 15yr (timing_analysis) standpoint, to facilitate PINT/Tempo2 comparisons, should we specify IERS2003 as an early check (I can open a quick MR if so) for timing models?

Currently PINT has no notion of an integer-valued parameter, so it's not as easy as changing their type. It might be possible to arrange for PINT to output parameter values that happen to be integers without the `.0`, which ought to resolve that problem.

I think - if I understand correctly - there is no particular negative to all NANOGrav timing solutions using `ECL IERS2003`; this will make the coordinates mean the same thing to all three programs and should not introduce any inaccuracy. This doesn't need PINT support, but would drop nicely into the same part of `timing_analysis` that makes sure we use `EPHEM DE440`. Like that, re-fitting would be necessary, so it would only take effect as we re-fit and check new par files into git.

**Comment by aarchiba** [comment]
Currently this produces par files PINT cannot read correctly as a result of #1019 .

**Comment by aarchiba** [comment]
We do have booleans, and I think that on read we accept 1/0 or Y/N; I'm not sure what we write. The Parameter types are... complicated, but adding an integer type might not be too bad. (PINT would want to accept expressions like 5.0 since it has been writing these.)

**Comment by davidnice** [comment]
Re: DMDATA being an integer, I fixed up Tempo a couple weeks ago so it can handle non-integer values.

Re: NHARM.  This is weird in Tempo right now.  Tempo reads it in as an integer to a temporary variable.  It prints an error if it is greater than 4.  It then does nothing else with it, i.e., it does not store it in a variable that gets used elsewhere.  Very strange!  Perhaps it is a legacy of an incomplete attempt to upgrade the ELL1H model (for which I believe NHARM is a control parameter).  It is simple enough to allow Tempo to treat it as a floating number (in fact, I will do that now), but really the parameter is just ignored.

**Comment by davidnice** [comment]
I think it is OK to use IERS2003 obliquity of the ecliptic.  This isn't the most up-to-date standard value, but it is hard-wired in Tempo2 (at least the last time I checked), so it would increase compatibility across packages if we use it.  It doesn't affect scientific results in any material way since its effect is only to perturb positions in ecliptic coordinates.

**Comment by demorest** [comment]
Making tempo input more robust is good, but I think it's mildly confusing for PINT to output floating point values for things that are actually integers or boolean, so I think that should still be fixed up.  Regarding NHARM, I forgot about that!  The tempo ELL1H model only implements the four-harmonic case (H3 and/or H4 params) which is a subset of what is available in tempo2.  That test is just there as a "safety feature" so that people don't get tripped up by this with no warning.

**Comment by aarchiba** [comment]
Should DMDATA not be a boolean? This does mean PINT will print out Y/N.

**Comment by aarchiba** [comment]
Now we have `intParameter` values, which accept floating-point values that are also integers, and output integers without decimal places.

**Comment by aarchiba** [comment]
> Making tempo input more robust is good, but I think it's mildly confusing for PINT to output floating point values for things that are actually integers or boolean, so I think that should still be fixed up. Regarding NHARM, I forgot about that! The tempo ELL1H model only implements the four-harmonic case (H3 and/or H4 params) which is a subset of what is available in tempo2. That test is just there as a "safety feature" so that people don't get tripped up by this with no warning.

By default. PINT outputs booleans as Y/N. It is not clear whether TEMPO can safely read these, since it prefers 1/0 for booleans.

**Comment by aarchiba** [comment]
Note to self: 

more integer paramters - `NTOA`

booleans (but are booleans safe?):  `CORRECT_TROPOSPHERE`, 'DMDATA', 'PLANET_SHAPIRO'

**Comment by demorest** [comment]
tempo only accepts 1/0 for booleans.  But it looks like the only boolean params it reads are DMDATA and DMXFIX.  Would not be hard to modify tempo to also take Y/N for these.

tempo2 accepts both 1/0 and Y/N on input, but writes Y/N in output par files.

**Comment by davidnice** [comment]
I have updated Tempo to allow flexible inputs of boolean parameter flags:
  -- "True" values are registered for strings starting with 'T' or 'Y' or 't' or 'y'
  -- "True" values are registered for numbers greater than zero, integer or real
  -- "False" values are registered for strings starting with 'F' or 'N' or 'f' or 'n'
  -- "False" values are registered for numbers zero and less

Tempo now uses this for:
-- DMXFIX
-- DMDATA
-- JUMP 
-- START
-- FINISH

Other boolean flags allowed by Tempo2/PINT flags (e.g., PLANET_SHAPIRO) are ignored by Tempo.

**Comment by aarchiba** [comment]
> I have updated Tempo to allow flexible inputs of boolean parameter flags:
> -- "True" values are registered for strings starting with 'T' or 'Y' or 't' or 'y'
> -- "True" values are registered for numbers greater than zero, integer or real
> -- "False" values are registered for strings starting with 'F' or 'N' or 'f' or 'n'
> -- "False" values are registered for numbers zero and less
> 
> Tempo now uses this for:
> -- DMXFIX
> -- DMDATA
> -- JUMP
> -- START
> -- FINISH
> 
> Other boolean flags allowed by Tempo2/PINT flags (e.g., PLANET_SHAPIRO) are ignored by Tempo.

Okay, I think we should be compatible with current TEMPO and TEMPO2 if files are written with `alias_translation=tempo_aliases`.`

**Comment by aarchiba** [comment]
Now supports remembering which alias was used on reading for par files. It would be good to add remembering which alias to use for observatories, though that's complicated because the input might use more than one alternative (so how do we pick which?). This should also affect observatory aliases in par files (TZRSITE), though handling that gracefully is not so easy.

**Comment by aarchiba** [comment]
This now allows you to set TOAs objects and on TimingModel objects into modes where they write themselves in TEMPO/TEMPO2-compatible formats. Otherwise TimingModel objects try to retain the way they were written in the input par files. TOAs objects don't, yet.

**Pull request merged**